core/src/main/java/hudson/model/View.java

```diff
@@ -1000,6 +1002,18 @@ public synchronized void doDoDelete(StaplerRequest req, StaplerResponse rsp) thr
      */
     public abstract Item doCreateItem( StaplerRequest req, StaplerResponse rsp ) throws IOException, ServletException;
 
+    /**
+     * An API method to get the allowed {$link TopLevelItem}s and its categories.
+     *
+     * @return A {@link Categories} entity that is shown as  JSON file.
+     *
+     * @since TODO
+     */
+    @Restricted(DoNotUse.class)
+    public Categories doCategories(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
+        return ItemGroupMixIn.getCategories(Jenkins.getAuthentication(), Jenkins.getInstance());
```

Comment:
Because when you create a new item inside a `Folder` (more specifically, inside a `View` of some `ItemGroup & ViewGroup` other than `Jenkins`), there is a specific list of `TopLevelItemDescriptor`s which are allowed, and this list can only be obtained by passing the correct `ItemGroup` to `Items.all`, and then passing the right `ItemGroup`/`ViewGroup` (or, preferably, `View`) to `DescriptorVisibilityFilter`.

If you try your current code with, say, `cloudbees-folders-plus`, you will see that it incorrectly offers item types which the original `View/newJob` did not.